backend/src/main/java/gov/cdc/usds/simplereport/api/patientExperience/PatientExperienceController.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import static gov.cdc.usds.simplereport.api.Translators.parseSymptoms;
+import static gov.cdc.usds.simplereport.api.Translators.parseEmail;
+import static gov.cdc.usds.simplereport.api.Translators.parseEthnicity;
+import static gov.cdc.usds.simplereport.api.Translators.parseGender;
+import static gov.cdc.usds.simplereport.api.Translators.parsePhoneNumber;
+import static gov.cdc.usds.simplereport.api.Translators.parseRace;
+import static gov.cdc.usds.simplereport.api.Translators.parseState;
+import static gov.cdc.usds.simplereport.api.Translators.parseString;
+
+import gov.cdc.usds.simplereport.api.exceptions.FeatureFlagDisabledException;
+import gov.cdc.usds.simplereport.api.exceptions.InvalidPatientLinkException;
+import gov.cdc.usds.simplereport.api.model.AoEQuestions;
+import gov.cdc.usds.simplereport.api.model.pxp.PxpApiWrapper;
+import gov.cdc.usds.simplereport.api.model.pxp.PxpPersonWrapper;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.model.PatientLink;
+import gov.cdc.usds.simplereport.db.model.Person;
+import gov.cdc.usds.simplereport.db.model.TestEvent;
+import gov.cdc.usds.simplereport.db.model.auxiliary.TestResult;
+import gov.cdc.usds.simplereport.service.PatientLinkService;
+import gov.cdc.usds.simplereport.service.PersonService;
+import gov.cdc.usds.simplereport.service.TestEventService;
+import gov.cdc.usds.simplereport.service.TestOrderService;
+
+@RestController
+@RequestMapping(value = "/pxp")
+@Validated
+public class PatientExperienceController {
+  private static final Logger LOG = LoggerFactory.getLogger(PatientExperienceController.class);
+
+  @Autowired
+  private PersonService ps;
+
+  @Autowired
+  private PatientLinkService pls;
+
+  @Autowired
+  private TestOrderService tos;
+
+  @Autowired
+  private TestEventService tes;
+
+  @Value("${feature-flags.patient-links:false}")
+  private boolean patientLinksEnabled;
+
+  /**
+   * Verify that the patient-provided DOB matches the patient on file for the
+   * patient link id. It returns the full patient object if so, otherwise it
+   * throws an exception
+   */
+  @PutMapping("/link/verify")
+  public PxpPersonWrapper getPatientLinkVerify(@RequestBody PxpApiWrapper<Void> body)
+      throws InvalidPatientLinkException {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    Person p = pls.getPatientLinkVerify(body.getPlid(), body.getDob());
+    TestEvent te = tes.getLastTestResultsForPatient(p);
+    return new PxpPersonWrapper(p, te);
+  }
+
+  @PutMapping("/patient")
+  public Person updatePatient(@RequestBody PxpApiWrapper<Person> body) throws InvalidPatientLinkException {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    pls.getPatientLinkVerify(body.getPlid(), body.getDob()); // gotta verify!
```

Comment:
This slightly scares me. Is it possible we can move the verification step into a `@PreAuthorize` annotation at the class level, or into a MessageConverter that will automatically get invoked when a PxpApiWrapper message body is involved?

Reply:
Good call! Took me a minute to figure out the right syntax for PreAuthorize, but this change is here: 65cbb8f